integration/coverage_test.go

```diff
@@ -112,17 +112,12 @@ var _ = Describe("Coverage Specs", func() {
 		os.RemoveAll("./_fixtures/combined_coverage_fixture/coverage.txt")
 	})
 
-	It("Creates directories in path if they don't exist", func() {
+	It("Fails with an error if cannot combine the suite coverage profiles because the output dir does not exist", func() {
```

Comment:
Could we possible put this in a `Describe`?
```
Describe("when the output dir does not exist", func() {
  It("fails with an error", func() {
```
I don't think the text "cannot combine the suite coverage profiles" adds very much, so I would omit it.

Reply:
The reason I specified "cannot combine the suite coverage profiles" is that by omitting the `-coverageprofile` flag, ginkgo will try to move the separate coverage profiles into the new directory without combining them.

As far as I understand, this is handled in another function (see [here](https://github.com/onsi/ginkgo/blob/master/ginkgo/run_command.go#L114), which has also different tests. 

How can we change the text to be clearer and explain that this test is specifically about `-coverageprofile` flag?

Reply:
```
Describe("when combining coverage profiles", func() {
  Describe("when the output dir does not exist", func() {
    It("fails with an error", func() {
```
A very long `It()` is a smell, but I'm not sure this is any better.

Reply:
As an aside from the general discussion, I'd suggest using `Context(` in the case where the thing under test is undergoing different preconditions rather than `Describe(`. In fact we also have `When(` which is an alias for `Context(` and prefixes `when` to the description.

Reply:
@blgm I have changed the description to be "Fails with an error if output dir and coverprofile were set, but the output dir did not exist", which follows the same structure to the `It` above.

In my opinion, adding a level of nesting such as `Context`, `When`, or `Describe` would make the tests less readable, but I agree that it was a very long `It` description.

What do you think about the message now?

Reply:
@nodo, I think the message is clear, and definitely good enough.  I take your point regarding nesting.  I think it can lead to clearer code, but it would be out of step with the rest of the file, and consistency is more important.